kola/tests/update/update.go

```diff
@@ -211,3 +222,86 @@ func splitNewlineEnv(envs string) map[string]string {
+	// b) stored on the OEM partition and gets moved to the rootfs and the new one is moved to the OEM partition
+	_ = c.MustSSH(noIgnition, fmt.Sprintf(`set -euxo pipefail
+sudo systemctl mask --now systemd-sysext
+sudo mkdir -p /etc/flatcar/sysext /etc/flatcar/oem-sysext /usr/share/oem/sysext /etc/extensions
+echo ID=test | sudo tee /usr/share/oem/oem-release
+echo myext | sudo tee /etc/flatcar/enabled-sysext.conf
+sudo touch /usr/share/oem/sysext/active-oem-test /etc/flatcar/oem-sysext/oem-test-%[1]s.raw /etc/flatcar/oem-sysext/oem-test-1.2.3.raw /etc/flatcar/sysext/flatcar-myext-%[1]s.raw
+sudo ln -fs /etc/flatcar/oem-sysext/oem-test-1.2.3.raw /etc/extensions/oem-test.raw
+`, version))
+	if err := noIgnition.Reboot(); err != nil {
+		c.Fatalf("couldn't reboot: %v", err)
+	}
+	// Check that the right symlinks are set up for case "a)" and prepare the next boot
+	_ = c.MustSSH(noIgnition, fmt.Sprintf(`set -euxo pipefail
+[ "$(readlink -f /etc/extensions/oem-test.raw)" = "/usr/share/oem/sysext/oem-test-%[1]s.raw" ] || { echo "OEM symlink wrong"; exit 1 ; }
+[ "$(readlink -f /etc/extensions/flatcar-myext.raw)" = "/etc/flatcar/sysext/flatcar-myext-%[1]s.raw" ] || { echo "Extension symlink wrong"; exit 1; }
+sudo mv /usr/share/oem/sysext/oem-test-%[1]s.raw /etc/flatcar/oem-sysext/
+sudo mv /etc/flatcar/oem-sysext/oem-test-1.2.3.raw /usr/share/oem/sysext/
+sudo ln -fs /usr/share/oem/sysext/oem-test-1.2.3.raw /etc/extensions/oem-test.raw
+`, version))
+	if err := noIgnition.Reboot(); err != nil {
+		c.Fatalf("couldn't reboot: %v", err)
+	}
+	// Check that the boot logic set up the right sysext symlinks for case "b)"
+	testCmds := fmt.Sprintf(`set -euxo pipefail
+[ "$(readlink -f /etc/extensions/oem-test.raw)" = "/usr/share/oem/sysext/oem-test-%[1]s.raw" ] || { echo "OEM symlink wrong"; exit 1 ; }
+[ "$(readlink -f /etc/extensions/flatcar-myext.raw)" = "/etc/flatcar/sysext/flatcar-myext-%[1]s.raw" ] || { echo "Extension symlink wrong"; exit 1; }
+`, version)
+	_ = c.MustSSH(noIgnition, testCmds)
+	noIgnition.Destroy()
+	// The second test case is to use Ignition and Ignition will also
+	// mount the OEM partition in the initrd
+	conf := conf.ContainerLinuxConfig(fmt.Sprintf(`storage:
+  filesystems:
+     - name: oem
+       mount:
+         device: "/dev/disk/by-label/OEM"
+         format: "btrfs"
+  files:
+    - path: /oem-release
+      filesystem: oem
+      contents:
+        inline: |
+          ID=test
+    - path: /sysext/active-oem-test
+      filesystem: oem
+    - path: /sysext/oem-test-%[1]s.raw
+      filesystem: oem
+    - path: /etc/flatcar/enabled-sysext.conf
+      contents:
+        inline: |
+          myext
+    - path: /etc/flatcar/sysext/flatcar-myext-%[1]s.raw
+systemd:
+  units:
+    - name: systemd-sysext.service
+      mask: true
+`, version))
+	withIgnition, err := c.NewMachine(conf)
+	if err != nil {
+		c.Fatalf("creating test machine: %v", err)
+	}
+	_ = c.MustSSH(withIgnition, testCmds)
```

Comment:
Yes, the Ignition case is different. I didn't want to do the same tests again each time with new VMs because the logic hit is the same, the main difference is that with Ignition the OEM mount point is set up a bit differently and we also just want to check that the logic is triggered at all because another systemd target unit is used to pull it in.

Reply:
I've added a longer comment on this